sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -113,21 +123,40 @@
                     .authority(authorityUrl);
             if (options.getProxyOptions() != null) {
                 applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
+            } else {
+                HttpPipeline pipeline = setupPipeline(request.getHttpClient(), request.getLogOptions());
+                applicationBuilder.httpClient(new HttpPipelineAdapter(pipeline));
             }
+
             if (options.getHttpPipeline() != null) {
                 applicationBuilder.httpClient(new HttpPipelineAdapter(options.getHttpPipeline()));
             }
```

Comment:
If `options.getHttpPipeline()` is not null, the client provided in `TokenRequestContext` is overwritten. Can you please add comments on why this is done? Also, it would be better to check if `options.getHttpPipeline()` is not null first instead of setting up the pipeline which might get overwritten immediately after.